streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java

```diff
@@ -78,6 +79,23 @@ public Headers headers() {
         return headers;
     }
 
+    public long sizeBytes() {
+        long size = 0L;
+        size += 8; // value.context.timestamp
+        size += 8; // value.context.offset
+        if (topic != null) {
+            size += topic.toCharArray().length;
```

Comment:
a `String` also store the length (it's a `char[]` internally) -- should we add 4 more bytes here?

Also, has a `char[]` similar overhead than a regular object?

Reply:
I believe arrays also store their types. But again, we are getting into JVM implementation details. There are too many JVM implementations for us to be expected to worry about this, IMHO.